### PR TITLE
KM290 ✅ Add remote state

### PR DESCRIPTION
**Pull request by deifyed**
## Description

- Adds S3 tooling
- Adds DynamoDB tooling
- Adds remote state by uploading and downloading the state.db file from S3
- Adds protection from mutating state asyncronously by using DynamoDB as a lock

## Motivation and Context

[KM290](https://trello.com/c/6CJ7xMz2)

## How to prove the effect of this PR?

1. Run the upgrade in [this](https://github.com/oslokommune/okctl-upgrade/pull/4) PR
2. Run `okctl apply cluster` on an existing cluster and verify that it doesn't create a new cluster

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the release notes (for the next release).

### CODE REVIEW

**Review by yngvark — APPROVED** [review]
Looks good, the hooks definitely makes things separated :+1:

File: pkg/client/core/api/rest/client_test.go [hunk 1]
```diff
@@ -71,3 +72,149 @@ func TestKindSanity(t *testing.T) {
...
+	testCases := []struct {
+		name string
+
+		withDebugFlag   bool
+		withRequestBody interface{}
+
+		expect string
+	}{
+		{
+			name:          "Should work produce nothing when debug is false",
```
[the remark this answers is not in the document]
Comment (deifyed) [reply pkg/client/core/api/rest/client_test.go:103]: :see_no_evil:

File: pkg/client/core/api/rest/client_test.go [hunk 2]
```diff
@@ -71,3 +72,149 @@ func TestKindSanity(t *testing.T) {
 
 	assert.False(t, errors.IsKind(err, errors.Decrypt))
 }
+
+type dummyBody struct {
+	Cat string `json:"cat"`
+	Dog string `json:"dog"`
+}
+
+type dummyCencoredRequest struct {
```
Comment (fredriv) [inline pkg/client/core/api/rest/client_test.go:81]: Typo: censored

Comment (deifyed) [reply pkg/client/core/api/rest/client_test.go:81]: :see_no_evil:

File: pkg/client/core/api/rest/client_test.go [hunk 3]
```diff
@@ -71,3 +72,149 @@ func TestKindSanity(t *testing.T) {
...
+
+			err := debugPrintRequest(meta, tc.withRequestBody)
+			assert.NoError(t, err)
+
+			assert.Equal(t, tc.expect, buf.String())
+		})
+	}
+}
+
+type dummyCencoredResponse struct {
```
Comment (fredriv) [inline pkg/client/core/api/rest/client_test.go:150]: "censored" here as well

**Review by yngvark — APPROVED** [review]
:+1:

**Review by yngvark — APPROVED** [review]
:+1:

**Pull request merged**